modules/manage/partials/iceberg/use-iceberg-catalogs.adoc

```diff
@@ -0,0 +1,106 @@
+
+See xref:reference:properties/cluster-properties.adoc[Cluster Configuration Properties] for the full list of cluster properties to configure for a catalog integration.
+
+=== Example REST catalog configuration
+
+For example, if you have Redpanda cluster configuration properties set to connect to a REST catalog:
+
+[,yaml]
+----
+iceberg_catalog_type: rest 
+iceberg_rest_catalog_endpoint: http://catalog-service:8181 
+iceberg_rest_catalog_client_id: <rest-connection-user>
+iceberg_rest_catalog_client_secret: <rest-connection-password>
+----
+
+And you use Apache Spark as a processing engine, configured to use a catalog named `streaming`:
+
+[,spark]
+----
+spark.sql.catalog.streaming = org.apache.iceberg.spark.SparkCatalog
+spark.sql.catalog.streaming.type = rest
+spark.sql.catalog.streaming.uri = http://catalog-service:8181
+# You may need to configure additional properties based on your object storage provider.
+# See https://iceberg.apache.org/docs/latest/spark-configuration/#catalog-configuration and https://spark.apache.org/docs/latest/configuration.html
+# For example, for AWS S3:
+# spark.sql.catalog.streaming.io-impl = org.apache.iceberg.aws.s3.S3FileIO
+# spark.sql.catalog.streaming.warehouse = s3://<bucket-name>/
+# spark.sql.catalog.streaming.s3.endpoint = http://<s3-uri>
+----
+
+NOTE: Redpanda recommends setting credentials in environment variables so Spark can securely access your Iceberg data in object storage. For example, for AWS, use `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+
+Using Spark SQL, you can query the Iceberg table directly by specifying the catalog name, the namespace, and the table name:
+
+[,sql]
+----
+SELECT * FROM streaming.redpanda.<table-name>;
+----
+
+The Iceberg table name is the name of your Redpanda topic. Redpanda puts the Iceberg table into a namespace called `redpanda`, creating the namespace if necessary. 
+
+The Spark engine can use the REST catalog to automatically discover the topic's Iceberg table. Depending on your processing engine, you may need to also create a table in the engine to point the data lakehouse to the table location in the catalog. For an example, see xref:manage:iceberg/redpanda-topics-iceberg-snowflake-catalog.adoc[].
+endif::[]
+
+== Integrate file-system based catalog (`object_storage`)
+
+By default, Iceberg topics use the file-system based catalog (config_ref:iceberg_catalog_type,true,properties/cluster-properties[`iceberg_catalog_type`] cluster configuration set to `object_storage`). Redpanda stores the table metadata in https://iceberg.apache.org/javadoc/1.5.0/org/apache/iceberg/hadoop/HadoopCatalog.html[HadoopCatalog^] format in the same object storage bucket or container as the data files.
+
+If using the `object_storage` catalog type, you provide the object storage URI of the table's metadata.json file to an Iceberg client so it can access the catalog and data files for your Redpanda Iceberg tables.
```

Comment:
```suggestion
If using the `object_storage` catalog type, you provide the object storage URI of the table's `metadata.json` file to an Iceberg client so it can access the catalog and data files for your Redpanda Iceberg tables.
```

Reply:
We should probably put a reminder note in here about the fact that this metadata.json file only point to a specific  table snapshot. Due to the limitations of the object storage catalog specification in Apache Iceberg, tables must be updated anytime a new snapshot is created using this catalog type (effectively, any time new data is written to the table). For more information, see the Apache Iceberg documentation.